libwild/src/version_script.rs

```diff
@@ -86,19 +91,44 @@ impl<'data> BasicMatchRules<'data> {
         mangled: bool,
     ) -> bool {
         // Early exit before we actually demangle the name.
-        if self.exact.is_empty() {
+        if self.exact.is_empty() && self.escaped_exact.is_empty() {
```

Comment:
This early exit seems to now be redundant because the demangling is now guarded below by `if !self.exact.is_empty()` and also by `if !self.escaped_exact.is_empty()`. Although depending on what you do with the next comment below, you might not need the changes to this method at all, so look at this comment last.